api/user/wire_selfRegistration.go

```diff
@@ -0,0 +1,22 @@
+package user
+
+import (
+	"github.com/devtron-labs/devtron/pkg/user"
+	"github.com/devtron-labs/devtron/pkg/user/repository"
+	"github.com/google/wire"
+)
+
+//depends on sql,
+//TODO integrate user auth module
+
+var SelfRegistrationWireSet = wire.NewSet(
```

Comment:
if self registration required in base stack (hyperion) this wire set also need to import inside cmd/external-app, also ensure router entry.